clients/lite-xl/tests/init_show_document_outcome_test.lua

```
@@ -0,0 +1,471 @@
+local selection_calls = {}
+---One shared root-view instance: init.lua reaches the editor's open_doc
+---through the core fake's root_view field and the core.rootview module.
+---Returns a docview carrying a minimal .doc so selection conversion and
+---set_selection behavior are provable end to end.
+local core_root_view = {}
+function core_root_view.open_doc(_, doc)
+  local docview = {
+    filename = doc.filename,
+    doc = {
+      lines = { "hello world" },
+      set_selection = function(self, line1, col1, line2, col2)
```

Luacheck (1.2.0)

[warning] unused argument 'self'

(W212)